geocoder/beam.cpp

```diff
@@ -0,0 +1,31 @@
+#include "geocoder/beam.hpp"
+
+#include "base/assert.hpp"
+
+#include <algorithm>
+
+namespace geocoder
+{
+Beam::Beam(size_t capacity) : m_capacity(capacity) { m_entries.reserve(m_capacity); }
+
+void Beam::Add(Key const & key, Value value)
+{
+  if (m_capacity == 0)
```

Comment:
Strongly suggest to mark this check as UNLIKELY

Reply:
ok
will do in a later PR